playlist_souffle/use_cases/souffle_playlist.py

```diff
@@ -0,0 +1,36 @@
+"""Module for the souffle_playlist use case."""
+
+from datetime import datetime as dt
+from playlist_souffle.definitions.playlist import Playlist
+import playlist_souffle.entities.souffler as souffler_entity
+import playlist_souffle.entities.track as track_entity
+
+def souffle_playlist(spotify, playlist_uri, user_id, shuffle_by):
+    """Souffle a playlist."""
+
+    original_playlist = Playlist(
+        owner=None,
+        name=spotify.fetch_playlist_name(playlist_uri),
```

Comment:
you could have a `fetch_playlist` function in your gateway that returns a `Playlist`